icu4c/source/i18n/calendar.cpp

```diff
@@ -841,7 +841,12 @@ Calendar::operator=(const Calendar &right)
         fWeekendCeaseMillis      = right.fWeekendCeaseMillis;
         fNextStamp               = right.fNextStamp;
         uprv_strncpy(validLocale, right.validLocale, sizeof(validLocale));
```

Comment:
Somehow a simple move down the line would solve the warning

--- a/icu4c/source/i18n/calendar.cpp
+++ b/icu4c/source/i18n/calendar.cpp
@@ -841,8 +841,8 @@ Calendar::operator=(const Calendar &right)
         fWeekendCeaseMillis      = right.fWeekendCeaseMillis;
         fNextStamp               = right.fNextStamp;
         uprv_strncpy(validLocale, right.validLocale, sizeof(validLocale));
-        uprv_strncpy(actualLocale, right.actualLocale, sizeof(actualLocale));
         validLocale[sizeof(validLocale)-1] = 0;
+        uprv_strncpy(actualLocale, right.actualLocale, sizeof(actualLocale));
         actualLocale[sizeof(validLocale)-1] = 0;
     }